exec/etherfilter.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2012-2018 Red Hat, Inc.  All rights reserved.
+ *
+ * Author: Fabio M. Di Nitto <fabbione@kronosnet.org>
+ *
+ * This software licensed under GPL-2.0+, LGPL-2.0+
+ */
+
+#include "config.h"
+
+#include <arpa/inet.h>
+#include <netinet/ether.h>
+#include <string.h>
+
+#include "etherfilter.h"
+
+/*
+ * stole from linux kernel/include/linux/etherdevice.h
+ */
+
+static inline int is_zero_ether_addr(const uint8_t *addr)
+{
+	return !(addr[0] | addr[1] | addr[2] | addr[3] | addr[4] | addr[5]);
+}
+
+static inline int is_multicast_ether_addr(const uint8_t *addr)
+{
+	return 0x01 & addr[0];
+}
```

Comment:
we can use BSD licensed implementation http://nixdoc.net/man-pages/NetBSD/man9/ETHER_IS_MULTICAST.9.html